api/src/main/java/jakarta/enterprise/inject/build/compatible/spi/Parameters.java

```diff
@@ -0,0 +1,88 @@
+package jakarta.enterprise.inject.build.compatible.spi;
+
+/**
+ * A {@code String}-keyed parameter map. The parameter mappings are defined
+ * by a synthetic component builder. The CDI container passes the map
+ * to functions defined by the same synthetic component builder, whenever
+ * it needs to invoke those functions. That is:
+ *
+ * <ul>
+ * <li>a synthetic bean {@linkplain SyntheticBeanCreator creation} and
+ *   {@linkplain SyntheticBeanDisposer destruction} function, defined by
+ *   {@link SyntheticBeanBuilder};</li>
+ * <li>a synthetic observer {@linkplain SyntheticObserver notification}
+ *   function, defined by {@link SyntheticObserverBuilder}.</li>
+ * </ul>
+ *
+ * Parameter values are transferred from the builder to the {@code Parameters}-accepting
+ * function without a change. For example, if the builder defines an {@code int} parameter,
+ * it must be looked up as {@code int} and cannot be looked up as {@code long}.
+ * <p>
+ * Values of primitive types may be looked up either using the primitive type (such as
+ * {@code int.class}), or using the corresponding wrapper type ({@code Integer.class}).
+ * The return value is always of the wrapper type.
+ * <p>
+ * Class-typed parameters are available as instances of {@link Class}, even if an instance
+ * of {@code ClassInfo} was passed to the builder.
+ * <p>
+ * Similarly, annotation-typed parameters are always available as instances of the annotation
+ * type, even if an instance of {@code AnnotationInfo} was passed to the builder.
+ */
+public interface Parameters {
```

Comment:
The original `Map<String, Object>`-based API isn't exactly pleasant when you've got more than just 2 or 3 parameters. So I thought a dedicated API would be better (and would have much smaller surface).

Not sure if this style (`get(String, Class)`) is better than the other suggested style (`getInt(String)`, `getString(String)`, `getEnum(String)`, `getAnnotation(String)` etc.).

Reply:
FWIW this is what we have in Micronaut that is equivalent https://github.com/micronaut-projects/micronaut-core/blob/19fef5801f31a729d88d8df47c3410fbfa442eda/core/src/main/java/io/micronaut/core/annotation/AnnotationValueResolver.java#L27